pkg/virt-operator/install-strategy/create.go

```diff
@@ -1415,10 +1419,18 @@ func generateServicePatch(kv *v1.KubeVirt,
 		deleteAndReplace = true
 	} else if cachedService.Spec.ClusterIP != service.Spec.ClusterIP {
 
-		// clusterIP is not mutable. A ClusterIP == "" will mean one is dynamically assigned.
-		// It is okay if the cached service has a ClusterIP and the target one is empty. We just
-		// ensure the cached ClusterIP is carried over in the Patch.
-		deleteAndReplace = true
+		if service.Spec.ClusterIP == "" {
```

Comment:
Why do we rely on ClusterIP?

Reply:
Service resources are a bit strange.  If you set the ClusterIP == "", then the k8s backend automatically assigns an IP address. The weird thing here for us is that the k8s backend actually mutates the service's spec with that new ClusterIP field.

So, the install strategy's desired service has `ClusterIP == ""` and the cached service has `ClusterIP == <some dynamic IP>`

Typically for our other resources we'd simply stomp on any changes that differ in the spec from our desired spec, versus what we see cached. However for a ClusterIP service, we actually need to preserve the cached ClusterIP field and merge it with the install strategies desired service spec.

Basically, cluster ip is something that we set to "" and magically turns into an IP later on. This inconsistency between our desired spec and what it gets mutated into requires some special handling.

Reply:
Thanks, that makes absolute sense :+1: